core/src/main/java/com/schibsted/spt/data/jslt/impl/BuiltinFunctions.java

```diff
@@ -1303,8 +1305,17 @@ public JsonNode call(JsonNode input, JsonNode[] arguments) {
         if(aURL.getUserInfo() != null && !aURL.getUserInfo().isEmpty())
           objectNode.put("userinfo", aURL.getUserInfo());
         return objectNode;
-      } catch (MalformedURLException | UnsupportedEncodingException e) {
-        throw new JsltException("Can't parse " + urlString, e);
+      } catch (Throwable e) {
+        if (throwOnFailure) {
+          throw new JsltException("Can't parse " + urlString, e);
+        } else {
+          final ObjectNode errorNode = NodeUtils.mapper.createObjectNode();
+          Class errorClass = e.getClass();
+          errorNode.put("error", errorClass.getSimpleName());
```

Comment:
Why not the full class name?

Reply:
I don't feel strongly over this. I can change to full class name.

Reply:
Changed to use `getName`